superx/routes/home.py

```diff
@@ -170,3 +174,20 @@ def city_search():
     """
     city_list = db.session.query(Branch.city).order_by(Branch.city).distinct().all()
     return render_template('city_list.html', city_list=city_list)
+
+
+def update_num_items():
+    """
+    renders number of items chosen for a product according to user's input
+    """
+    id_to_update = request.form.get('id')
+    num_items = request.form.get('num_items')
+
+    cart_list = session['cart']
+    for i in range(len(cart_list)):
+        if cart_list[i]['id'] == id_to_update:
+            cart_list[i]['num_items'] = num_items
+            break
```

Comment:
this could be cleaner. Two options:
1) don't use the list id, iterate through the list itself.
```
for item in cart_list:
    if item["id"] == id_to_update:
        item["num_items'] = num_items
        break

```
2) instead of have you cart be a list of dictionaries, make it a dictionary of dictionaries, so that you can always go straight to the right element instead of having to search:
`cart_list[id_to_update]["num_items"] = num_items`

Reply:
that's true, cleaned it with your advice